java/org/contikios/cooja/MoteInterfaceHandler.java

```diff
@@ -105,6 +105,7 @@ public MoteInterfaceHandler(Mote mote, Class<? extends MoteInterface>[] interfac
         addInterface(intf);
       } else {
         logger.fatal("Could not load interface: " + interfaceClass);
+        throw new RuntimeException("Could not load interface " + interfaceClass);
```

Comment:
It will be a somewhat more significant change, but what do you think about `MoteInterface.generateInterface()` throwing an exception with information about what went wrong instead of simply returning `NULL`? It will require making sure the call path to it also throws `MoteTypeCreationException`, but I think it could provide much better error information. Also, it would remove the need of catching an exception in Simulation since it would already be the correct exception type.

Reply:
Updated the patch with that.